### DISCUSSION TITLE
Totale dei pagamenti errati, ma in realtà è corretto

### DISCUSSION THREAD

**Discussion by antonelloventurino** [fattureincloud/api#173] [Q&A] [answered]
Salve,
ho un problema strano con il totale dei pagamenti. Sto inviando i prezzi iva inclusa, ho solo due articoli con prezzi 19.9 € e 12.8 € e quindi il totale è 32.7 €

Il JSON è il seguente:

```
{
  "data": {
    "entity": {
      "type": "person",
      "country": "Italia",
      "e_invoice": false,
      "name": "Tizio Caio",
      "first_name": "Tizio",
      "last_name": "Caio",
      "vat_number": "",
      "tax_code": "",
      "address_street": "Via Prova 11",
      "address_postal_code": "10100",
      "address_city": "Roma",
      "address_province": "RM",
      "email": ["posta@posta.it"](mailto:posta@posta.it),
      "phone": "",
      "fax": "",
      "certified_email": "",
      "ei_code": ""
    },
    "type": "invoice",
    "numeration": "",
    "currency": {
      "id": "EUR",
      "exchange_rate": null
    },
    "language": {
      "code": "it"
    },
    "payment_method": {
      "name": "359583"
    },
    "e_invoice": false,
    "use_split_payment": false,
    "use_gross_prices": true,
    "notes": "Note",
    "date": "2022-11-16",
    "items_list": [
      {
        "code": "codice",
        "name": "Nome Prodotto",
        "qty": "1",
        "gross_price": "19.9000",
        "not_taxable": false,
        "stock": true,
        "vat": {
          "id": "9"
        }
      },
      {
        "code": "",
        "name": "Spedizione",
        "qty": 1,
        "gross_price": "12.8000",
        "not_taxable": false,
        "stock": false,
        "vat": {
          "id": "9"
        }
      }
    ],
    "payments_list": [
      {
        "due_date": "2022-11-16",
        "amount": "32.7000",
        "status": "paid",
        "paid_date": "2022-11-16",
        "payment_account": {
          "id": "49080"
        }
      }
    ],
    "show_totals": "all",
    "show_paypal_button": false,
    "show_notification_button": false,
    "accompanying_invoice": false
  }
}
```

Come mai ricevo il seguente messaggio: 
_Il totale dei pagamenti non corrisponde al totale da pagare_

Grazie

**Comment by emajo** [comment, marked as answer]
Ciao @antonelloventurino, ho provato a crere un documento con il tuo json e viene creato correttamente, potrebbe essere qualche impostazione legata al tuo account, ti consiglio di fare la [chiamata per i totali](https://developers.fattureincloud.it/docs/guides/invoice-totals) che ti mostrerà il totale e le relative parti di cui è composto.

Grazie,
Emanuele.

**Reply by antonelloventurino** [reply]
Ciao @emajo, grazie per la risposta.
Provando con la chiamata per i totali mi viene restituito questo:

`{
  "data": {
    "amount_net": 32.7,
    "amount_global_cassa_taxable": 32.7,
    "taxable_amount": 32.7,
    "vat_list": {
      "0": {
        "amount_net": 32.7,
        "amount_vat": 0
      }
    },
    "amount_vat": 0,
    "amount_gross": 26.24,
    "amount_withholding_tax_taxable": 16.35,
    "amount_withholding_tax": 3.76,
    "amount_enasarco_taxable": 32.7,
    "amount_other_withholding_tax_taxable": 32.7,
    "amount_other_withholding_tax": 2.7,
    "amount_due": 26.24,
    "is_enasarco_maximal_exceeded": false,
    "amount_due_discount": null,
    "payments_sum": 32.7
  }
}`

Mentre su un altro account questo:
`{
  "data": {
    "amount_net": 32.7,
    "amount_global_cassa_taxable": 32.7,
    "taxable_amount": 32.7,
    "vat_list": {
      "0": {
        "amount_net": 32.7,
        "amount_vat": 0
      }
    },
    "amount_vat": 0,
    "amount_gross": 32.7,
    "amount_enasarco_taxable": 0,
    "amount_due": 32.7,
    "amount_due_discount": null,
    "payments_sum": 32.7
  }
}`

Il secondo non mi da errore, il primo sì, ma non riesco a capire come mai nel primo ci siano più campi.

**Reply by emajo** [reply]
Ciao @antonelloventurino, nel primo `amount_due` e `payments_sum` differiscono, perciò viene dato errore, i campi in più sono riferiti alle ritenute che non sono state invece settate nel secondo esempio.

Questi due risultati sono stati ottenuti con lo stesso body di partenza? Vedendo le differenze tra i due sembrerebbe che anche le richieste differiscano, non sembrerebbe un problema legato al cambio di account.

Grazie,
Emanuele.

**Reply by antonelloventurino** [reply]
È questo che non riesco a capire, perché i due risultati sono ottenuti con lo stesso JSON di partenza (quello del mio primo messaggio) ma su diversi account. L'unica cosa che ho cambiato è il `payment_account` altrimenti dava errore per ovvi motivi. 

Sembra, appunto, che l'errore sia account dipendente, ma non riesco a capire quale sia la causa.

**Reply by emajo** [reply]
Ciao @antonelloventurino, ti consiglio di scrivere al nostro supporto clienti indicando i due company id con cui stai facendo le chiamate così da andare a controllare più nel dettaglio la problematica riscontrata.

Grazie,
Emanuele.

**Reply by antonelloventurino** [reply]
Va bene, grazie. Gentilissimo

**Reply by valmoz** [reply]
Buongiorno.
è stata risolta questa problematica?

Grazie
Mauro

**Reply by antonelloventurino** [reply]
Salve, sì, ho risolto grazie.
